user_manager/api/v1/views.py

```diff
@@ -257,25 +326,75 @@ def perform_create(self, serializer):
         """
         Use serializer to create ``UserManagerRole`` object using provided data.
         """
-
         manager_id = self.kwargs['username']
-        email = serializer.validated_data.get('user', {}).get('email')
+
+        if not self.bulk_operation:
+            email = serializer.validated_data.get('user', {}).get('email')
+            username = serializer.validated_data.get('user', {}).get('username')
+
+            data = self.process_report(manager_id, username=username, email=email)
+            return serializer.create(data)
+
+        errors = []
+        reports = []
+        for serialized_report in serializer.validated_data:
+            email = serialized_report.get('user', {}).get('email')
+            username = serialized_report.get('user', {}).get('username')
+            try:
+                data = self.process_report(manager_id, username=username, email=email)
+                reports.append(data)
+            except NotFound as e:
+                identifier = email or username
+                errors.append({
+                    identifier: e.detail
+                })
+
+        created_reports = serializer.create(reports)
+        if errors:
+            raise NotFound(detail={
+                'created': {
+                    'count': len(created_reports),
+                    'results': created_reports,
+                },
+                'errors': errors
+            })
+
+        return created_reports
```

Comment:
I just did some manual testing and realised that the response format is very different in the case with and without an error. If there is no error, the response is just a simple list of objects of emails. However, in case of an error, the structure includes the breakdown or created and errors. Could you make this uniform? 

i.e. whether there is an error or not the response should be:
```json
{
  'created': {
     'count': x, 
     'results: ...,
   },
   'errors': [...]
}]
```
The errors key can be absent or empty in case there are no errors.